Source/Core/DolphinNoGUI/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+if(APPLE)
+	set(DOLPHIN_EXE_BASE Dolphin)
+else()
+	set(DOLPHIN_EXE_BASE dolphin-emu)
+endif()
+
+set(NOGUI_SRCS MainNoGUI.cpp)
+
+list(APPEND LIBS core uicommon cpp-optparse)
+
+if(USE_X11 OR ENABLE_HEADLESS)
+	set(DOLPHIN_NOGUI_EXE ${DOLPHIN_EXE_BASE}-nogui)
+	add_executable(${DOLPHIN_NOGUI_EXE} ${NOGUI_SRCS})
```

Comment:
I'm not sure what you're suggesting, probably because I don't understand cmake well enough. Which line is "changing the target name"? I don't see any variables being set more than once.